tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -54,29 +58,105 @@ struct SynthOrdered {
     }
 };
 
-template <class T>
-inline constexpr bool is_pair = false;
-template <class A, class B>
-inline constexpr bool is_pair<std::pair<A, B>> = true; // TRANSITION, std::pair spaceship not yet implemented
+struct OrderedChar {
+    OrderedChar() = default;
+    OrderedChar(const char c) : c(c){};
 
-template <class Container>
-void ordered_containers_test(const Container& smaller, const Container& smaller_equal, const Container& larger) {
+    OrderedChar& operator=(const char& other) {
+        this->c = other;
+        return *this;
+    }
+
+    operator char() {
+        return c;
+    }
+
+    char c;
+};
+
+auto operator<=>(const OrderedChar& left, const OrderedChar& right) {
+    return left.c <=> right.c;
+}
+
+auto operator==(const OrderedChar& left, const OrderedChar& right) {
+    return left.c == right.c;
+}
+
+struct WeaklyOrderedChar : public OrderedChar {};

```

Comment:
In product code, we use public inheritance implicitly with `struct` definitions. (No change requested - our conventions are looser for test code - but that may change soon.)